### ISSUE TITLE
[Rule] Unnecessary string conversion in conditions

### ISSUE THREAD

**Issue by bhirsz** [MarketSquare/robotframework-robocop#442]
A lot of people are not aware how evaluating condition works in Robot Framework (it uses Python) and create following conditions:
```
IF    r'''${value}''' == 'True'
    Keywords
END
```
instead of using value directly:
```
IF    ${value}
    Keywords
END
```
It could be tricky in some cases that's why we can work on this rule over the releases (extend for new cases).

**Comment by bhirsz** [comment]
Update - you can't use variable directly because it will insert the value of the variable as name. 

For example:
```
*** Variables ***
${VAR}    value

*** Test Cases ***
Test
    IF    ${VAR}    RETURN
```

It will evaluate to "if value" and fail because "value" variable doesn't exist. That's why we should propose $value syntax instead. 
It could be hit or miss rule (I'm not entirely sure it will not have false positives) that's why it could be community rule? 

The algorithm would:

1. Use iterate_variables (or similar, to check text before/after variable too) to find variables in the IF/WHILE condition
2. If variable is enclosed with " or ', propose $value syntax

**Comment by mnojek** [comment]
If you are concerned that the rule will not be 100% bulletproof, I think it should not end as a built-in rule.
In what circumstances you expect here to have false positives?

**Comment by bhirsz** [comment]
You will be not happy with my answer :D It's my gut feeling. I saw a lot of different ways of writing conditions in Robot Framework and I was afraid some of those were intentional string conversions that would be catched by this rule. 

However I can create this rule and test it on the different commercial projects to see if there is any issue with this rule. If there will be no issue it could be built-in rule, if there are some false positives that cannot be fixed easily it will be community rule.

**Comment by bhirsz** [comment]
I need to check where conditions are evaluated in Robot Framework and support them in this rule: Set Variable If, Evaluate etc. I can skip syntax that will be deorecated (like Continue For Loop If)

**Comment by bhirsz** [comment]
Conditions needs to be checked in:

IF
While
Pass Execution If
Set Variable If (note that it can chain multiple conditions)
Should Be True
Should Not Be True
Skip If

The rest is either deprecated or doesn't check for truth values (Evaluate evaluates so it can be any operation)
